File: Chapter11/C#/SystemBrowser/Coordinators/CoordinatorShutdown.cs
using System;
using System.Threading;

namespace SystemBrowser
{
  public class CoordinatorShutdown
  {
    // shortcut property
    Builder Builder
    {
      get {return Builder.Singleton;}
    }

    public void Run()
    {
      // disable the user interface during shutdown
      if (Builder.formMain != null)
        Builder.formMain.Visible = false;

      Builder.userSettings.Save();
    }
  }
}

File: Chapter11/VB/SystemBrowser/Coordinators/CoordinatorShutdown.vb
Public Class CoordinatorShutdown
  ' shortcut property
  ReadOnly Property Builder() As Builder
    Get
      Return Builder.Singleton
    End Get
  End Property

  Public Sub Run()
    ' disable the user interface during shutdown
    If Not Builder._formMain Is Nothing Then
      Builder._formMain.Visible = False
    End If

    Builder._userSettings.Save()
  End Sub
End Class